src/main/java/org/truffleruby/language/dispatch/UncachedDispatchNode.java

```diff
@@ -60,6 +61,10 @@ public Object executeDispatch(
             Object name,
             DynamicObject blockObject,
             Object[] argumentsObjects) {
+        if (RubyGuards.isForeignObject(receiverObject)) {
+            throw new UnsupportedOperationException();
+        }
```

Comment:
That's an extra cost for UncachedDispatchNode which is actually used in benchmarks.
At the very least the throw should be behind a transferToInterpreterAndInvalidate().
I think we should move this to the method lookup slow-path as that's not supported on foreign objects.